Source/CarthageKitTests/BuildArgumentsSpec.swift

```diff
@@ -0,0 +1,327 @@
+							"CODE_SIGN_IDENTITY="
+						]
+					}
+				}
+			}
+
+			context("when created with a project file") {
+				let project = ProjectLocator.ProjectFile(NSURL(string: "file:///Foo/Bar/project.xcodeproj")!)
+
+				it("has a default set of arguments specifying the project file") {
+					let subject = BuildArguments(project: project)
+
+					expect(subject.arguments) == [
+						"xcodebuild",
+						"-project",
+						"/Foo/Bar/project.xcodeproj",
+						"CODE_SIGNING_REQUIRED=NO",
+						"CODE_SIGN_IDENTITY="
+					]
+				}
+
+				it("includes the scheme if one is given") {
+					var subject = BuildArguments(project: project)
+					subject.scheme = "exampleScheme"
+
+					expect(subject.arguments) == [
+						"xcodebuild",
+						"-project",
+						"/Foo/Bar/project.xcodeproj",
+						"-scheme",
+						"exampleScheme",
+						"CODE_SIGNING_REQUIRED=NO",
+						"CODE_SIGN_IDENTITY="
+					]
+				}
+
+				it("includes the configuration if one is given") {
+					var subject = BuildArguments(project: project)
+					subject.configuration = "exampleConfiguration"
+
+					expect(subject.arguments) == [
+						"xcodebuild",
+						"-project",
+						"/Foo/Bar/project.xcodeproj",
+						"-configuration",
+						"exampleConfiguration",
+						"CODE_SIGNING_REQUIRED=NO",
+						"CODE_SIGN_IDENTITY="
+					]
+				}
+
+				describe("specifying the sdk") {
+					it("includes the sdk if it's not .MacOSX") {
+						for sdk: SDK in [.iPhoneOS, .iPhoneSimulator, .watchOS, .watchSimulator, .tvOS, .tvSimulator] {
+							var subject = BuildArguments(project: project)
+							subject.sdk = sdk
+
+							expect(subject.arguments) == [
+								"xcodebuild",
+								"-project",
+								"/Foo/Bar/project.xcodeproj",
+								"-sdk",
+								sdk.rawValue,
+								"CODE_SIGNING_REQUIRED=NO",
+								"CODE_SIGN_IDENTITY="
+							]
+						}
+					}
+
+					it("does not include the sdk flag if .MacOSX is specified") {
+						var subject = BuildArguments(project: project)
+						subject.sdk = .MacOSX
+
+						// Passing in -sdk macosx appears to break implicit dependency
+						// resolution (see Carthage/Carthage#347).
+						//
+						// Since we wouldn't be trying to build this target unless it were
+						// for OS X already, just let xcodebuild figure out the SDK on its
+						// own.
+						expect(subject.arguments) == [
+							"xcodebuild",
+							"-project",
+							"/Foo/Bar/project.xcodeproj",
+							"CODE_SIGNING_REQUIRED=NO",
+							"CODE_SIGN_IDENTITY="
+						]
+					}
+				}
+
+				it("includes the destination if given") {
+					var subject = BuildArguments(project: project)
+					subject.destination = "exampleDestination"
+
+					expect(subject.arguments) == [
+						"xcodebuild",
+						"-project",
+						"/Foo/Bar/project.xcodeproj",
+						"-destination",
+						"exampleDestination",
+						"CODE_SIGNING_REQUIRED=NO",
+						"CODE_SIGN_IDENTITY="
+					]
+				}
+
+				describe("specifying onlyActiveArchitecture") {
```

Comment:
This test is repeated from above. Can you extract the common bits into functions?